### ISSUE TITLE
XLA JIT compile failing on TF 2.9.2, A100 GPUs (on GCP VMs)

### ISSUE THREAD

**Issue by vivekex** [tensorflow/tensorflow#58034]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Bug

### Source

binary

### Tensorflow Version

2.9.2

### Custom Code

Yes

### OS Platform and Distribution

Debian GNU/Linux 10 (buster)

### Python version

3.7.12

### CUDA/cuDNN version

11.3/8

### GPU model and memory

A100, 80GB

### Current Behaviour?

```shell
Till a few weeks ago, I was able to successfully use XLA compilation when training with GPUs on GCP VMs.
Specifically, I used `compile_jit=True` when doing a model.compile() for a TF Keras model. Having trained my models, I didn't run any TF training jobs for a while. Recently (a week ago) I tried running the same models, but the code now complains that it cannot find libcudnn.so.8. The same code works without `compile_jit=True`, but that does require a slightly smaller batch size to fit in memory.

The log with `compile_jit=True` says "Could not load library libcudnn_cnn_train.so.8", but that library is most definitely present, since the same code does work without using XLA JIT compilation.
```

### Standalone code to reproduce the issue

```shell
Unable to provide a small case, haven't been able to reproduce on small models.
```

### Relevant log output

```shell
Could not load library libcudnn_cnn_train.so.8. Error: /opt/conda/bin/../lib/./libcudnn_ops_train.so.8: symbol _Z22cudnnGenericOpTensorNdILi3EE13cudnnStatus_tP12cudnnContext16cudnnGeneric
Op_t21cudnnNanPropagation_tPKdPKvPK17cudnnTensorStructS8_S8_SB_S8_S8_SB_Pv version libcudnn_ops_infer.so.8 not defined in file libcudnn_ops_infer.so.8 with link time reference
Could not load library libcudnn_cnn_train.so.8. Error: /opt/conda/bin/../lib/./libcudnn_ops_train.so.8: symbol _Z22cudnnGenericOpTensorNdILi3EE13cudnnStatus_tP12cudnnContext16cudnnGeneric
Op_t21cudnnNanPropagation_tPKdPKvPK17cudnnTensorStructS8_S8_SB_S8_S8_SB_Pv version libcudnn_ops_infer.so.8 not defined in file libcudnn_ops_infer.so.8 with link time reference
Could not load library libcudnn_cnn_train.so.8. Error: /opt/conda/bin/../lib/./libcudnn_ops_train.so.8: symbol _Z22cudnnGenericOpTensorNdILi3EE13cudnnStatus_tP12cudnnContext16cudnnGeneric
Op_t21cudnnNanPropagation_tPKdPKvPK17cudnnTensorStructS8_S8_SB_S8_S8_SB_Pv version libcudnn_ops_infer.so.8 not defined in file libcudnn_ops_infer.so.8 with link time reference
Could not load library libcudnn_cnn_train.so.8. Error: /opt/conda/bin/../lib/./libcudnn_ops_train.so.8: symbol _Z22cudnnGenericOpTensorNdILi3EE13cudnnStatus_tP12cudnnContext16cudnnGeneric
Op_t21cudnnNanPropagation_tPKdPKvPK17cudnnTensorStructS8_S8_SB_S8_S8_SB_Pv version libcudnn_ops_infer.so.8 not defined in file libcudnn_ops_infer.so.8 with link time reference
Could not load library libcudnn_cnn_train.so.8. Error: /opt/conda/bin/../lib/./libcudnn_ops_train.so.8: symbol _Z22cudnnGenericOpTensorNdILi3EE13cudnnStatus_tP12cudnnContext16cudnnGeneric
Op_t21cudnnNanPropagation_tPKdPKvPK17cudnnTensorStructS8_S8_SB_S8_S8_SB_Pv version libcudnn_ops_infer.so.8 not defined in file libcudnn_ops_infer.so.8 with link time reference
Please make sure libcudnn_cnn_train.so.8 is in your library path!
Please make sure libcudnn_cnn_train.so.8 is in your library path!
Please make sure libcudnn_cnn_train.so.8 is in your library path!
Please make sure libcudnn_cnn_train.so.8 is in your library path!
Please make sure libcudnn_cnn_train.so.8 is in your library path!
Please make sure libcudnn_cnn_train.so.8 is in your library path!
Please make sure libcudnn_cnn_train.so.8 is in your library path!


  (0) UNKNOWN:  Failed to determine best cudnn convolution algorithm for:
%cudnn-conv-bw-filter = (f32[1,1,768,1664]{1,0,2,3}, u8[0]{0}) custom-call(f32[2,768,90,180]{3,2,1,0} %add.19138, f32[2,1664,90,180]{3,2,1,0} %multiply.19168), window={size=1x1}, dim_labe
ls=bf01_01io->bf01, custom_call_target="__cudnn$convBackwardFilter", metadata={op_type="Conv2DBackpropFilter" op_name="gradient_tape/model/decode/Conv2D_1/Conv2DBackpropFilter" source_fil
e="/opt/conda/lib/python3.7/site-packages/keras/optimizers/optimizer_experimental/optimizer.py" source_line=176}, backend_config="{\"conv_result_scale\":1,\"activation_mode\":\"0\",\"side
_input_scale\":0}"

Original error: INTERNAL: All algorithms tried for %cudnn-conv-bw-filter = (f32[1,1,768,1664]{1,0,2,3}, u8[0]{0}) custom-call(f32[2,768,90,180]{3,2,1,0} %add.19138, f32[2,1664,90,180]{3,2
,1,0} %multiply.19168), window={size=1x1}, dim_labels=bf01_01io->bf01, custom_call_target="__cudnn$convBackwardFilter", metadata={op_type="Conv2DBackpropFilter" op_name="gradient_tape/mod
el/decode/Conv2D_1/Conv2DBackpropFilter" source_file="/opt/conda/lib/python3.7/site-packages/keras/optimizers/optimizer_experimental/optimizer.py" source_line=176}, backend_config="{\"con
v_result_scale\":1,\"activation_mode\":\"0\",\"side_input_scale\":0}" failed. Falling back to default algorithm.  Per-algorithm errors:

To ignore this failure and try to use a fallback algorithm (which may have suboptimal performance), use XLA_FLAGS=--xla_gpu_strict_conv_algorithm_picker=false.  Please also file a bug for the root cause of failing autotuning.
         [[{{node while/replica_3/StatefulPartitionedCall}}]]
         [[while/loop_body_control/_3358/_36]]
```
</details>

**Comment by vivekex** [comment]
Minimal reproducible exampel:


```
import tensorflow as tf


class Model(tf.keras.Model):

  def __init__(self):
    super().__init__()
    self.blocks = [
        tf.keras.layers.Conv2D(filters=32,
                               kernel_size=(3, 3),
                               strides=(1, 1),
                               padding='same',
                               data_format='channels_first',
                               name='upscale')
    ]

  def call(self, inputs):
    layer = inputs
    for b in self.blocks:
      layer = b(layer)
    return layer


def main():
  m = Model()
  x = [tf.random.uniform(shape=(32,90,180)) for _ in range(32)]
  y = [tf.random.uniform(shape=(32,90,180)) for _ in range(32)]
  ds = tf.data.Dataset.from_tensor_slices((x,y)).batch(16)
  m.compile(jit_compile=True, loss='mse')
  m.fit(ds, epochs=5)

if __name__ == '__main__':
  main()
```

This is on a VM created on GCP, whose details shown on login are:
```
======================================
Welcome to the Google Deep Learning VM
======================================

Version: tf2-gpu.2-9.m97
Based on: Debian GNU/Linux 10 (buster) (GNU/Linux 4.19.0-22-cloud-amd64 x86_64\n)

Resources:
 * Google Deep Learning Platform StackOverflow: https://stackoverflow.com/questions/tagged/google-dl-platform
 * Google Cloud Documentation: https://cloud.google.com/deep-learning-vm
 * Google Group: https://groups.google.com/forum/#!forum/google-dl-platform

To reinstall Nvidia driver (if needed) run:
sudo /opt/deeplearning/install-driver.sh
TensorFlow comes pre-installed with this image. To install TensorFlow binaries in a virtualenv (or conda env),
please use the binaries that are pre-built for this image. You can find the binaries at
/opt/deeplearning/binaries/tensorflow/
If you need to install a different version of Tensorflow manually, use the common Deep Learning image with the
right version of CUDA

Linux ultrag 4.19.0-22-cloud-amd64 #1 SMP Debian 4.19.260-1 (2022-09-29) x86_64

The programs included with the Debian GNU/Linux system are free software;
the exact distribution terms for each program are described in the
individual files in /usr/share/doc/*/copyright.

Debian GNU/Linux comes with ABSOLUTELY NO WARRANTY, to the extent
permitted by applicable law.
Last login: Mon Oct 10 04:29:08 2022 from 135.180.175.203
```

Driver version from `nvidia-smi`:
NVIDIA-SMI 510.47.03    Driver Version: 510.47.03    CUDA Version: 11.6

**Comment by vivekex** [comment]
This seems to not happen with tf-2.7 on the same machine, run in a docker container:

```
docker run --runtime=nvidia --name=gpu -it -d -w /home/doc -v /home/doc/:/home/doc/ gcr.io/deeplearning-platform-release/tf-gpu.2-7 /bin/bash
```

**Comment by vivekex** [comment]
Yes, I tried that flag, unfortunately it didn't change anything :(

**Comment by mohantym** [comment]
Ok @vivekex !
Thanks for the update. Could you look at this issue.
Thank you!

**Comment by sachinprasadhs** [comment]
Could you try matching the. below published configuration and let us know the outcome. Thanks!
<h4 id="gpu" data-text="GPU" role="presentation" style="box-sizing: inherit; margin: 32px 0px 16px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-variant-numeric: ; font-variant-east-asian: ; font-weight: ; font-stretch: ; font-size: 16px; line-height: ; font-family: Roboto, &quot;Noto Sans&quot;, &quot;Noto Sans JP&quot;, &quot;Noto Sans KR&quot;, &quot;Noto Naskh Arabic&quot;, &quot;Noto Sans Thai&quot;, &quot;Noto Sans Hebrew&quot;, &quot;Noto Sans Bengali&quot;, sans-serif; letter-spacing: normal; overflow: hidden; text-overflow: ellipsis; margin-inline-end: -40px; padding-inline-end: 40px; color: rgb(32, 33, 36); orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><span class="devsite-heading" role="heading" aria-level="4" style="box-sizing: inherit;">GPU</span><button type="button" class="devsite-heading-link button-flat material-icons" aria-label="Copy link to this section: GPU" data-title="Copy link to this section: GPU" data-id="gpu" style="box-sizing: border-box; appearance: none; background: 0px center; border: 0px; border-radius: var(--devsite-button-border-radius,2px); box-shadow: none; color: var(--devsite-icon-color,var(--devsite-secondary-text-color)); cursor: pointer; display: inline-block; font-style: normal; font-variant-ligatures: ; font-variant-caps: ; font-variant-numeric: ; font-variant-east-asian: ; font-weight: normal; font-stretch: ; font-size: 24px; font-family: &quot;Material Icons&quot;; height: 24px; letter-spacing: normal; line-height: 1; margin: var(--devsite-button-margin,0); margin-inline-end: var(--devsite-button-margin-x-end); max-width: var(--devsite-button-max-width,none); min-width: 36px; outline: 0px; overflow: hidden; padding: 0px 8px; text-align: center; text-decoration: none; text-overflow: ellipsis; text-transform: none; transition: background-color 0.2s ease 0s, border 0.2s ease 0s, box-shadow 0.2s ease 0s; vertical-align: bottom; white-space: nowrap; width: var(--devsite-button-width,auto); overflow-wrap: normal; direction: ltr; font-feature-settings: &quot;liga&quot;; -webkit-font-smoothing: antialiased; opacity: 0;"></button></h4><div class="devsite-table-wrapper" style="box-sizing: inherit; margin: var(--devsite-table-margin,16px 0); padding: 0px; overflow: auto; color: rgb(32, 33, 36); font-family: Roboto, &quot;Noto Sans&quot;, &quot;Noto Sans JP&quot;, &quot;Noto Sans KR&quot;, &quot;Noto Naskh Arabic&quot;, &quot;Noto Sans Thai&quot;, &quot;Noto Sans Hebrew&quot;, &quot;Noto Sans Bengali&quot;, sans-serif; font-size: 16px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">

Version | Python version | Compiler | Build tools | cuDNN | CUDA
-- | -- | -- | -- | -- | --
tensorflow-2.10.0 | 3.7-3.10 | GCC 9.3.1 | Bazel 5.1.1 | 8.1 | 11.2

</div>

**Comment by vivekex** [comment]
I _believe_ I was able to try that with a close enough setup. Getting sysdetails from TF:
```
import tensorflow as tf
sys_details = tf.sysconfig.get_build_info()
```
shows TF 2.10, python 3.7, CUDA 11.2, cudNN 8

This did let the code run successfully. I'm still unclear if this is a TF issue or a VM setup issue (I'm inclined to think it's a VM setup issue...)

**Comment by sachinprasadhs** [comment]
It could be due to the VM setup issue, It is always suggested to use the tested and published  build configurations.
 If your issue is resolved, could you please close the issue. Thanks!

**Comment by vivekex** [comment]
I'm able to replicate a working configuration using a clean Ubuntu VM
image, and using Conda to install the nvidia deps.

The problem still arises when using out-of-the-box TF Deep Learning images
that Google Cloud provides, so I suspect even just relying on pre-built
configurations is not sufficient, there's some other incompatibility. Will
close this bug, though, since it looks like a nVidia setup issue in the VM
and not TF issue.
